File: aspnet-with-ef/HuHao.BlogService.Domain/Core/Exceptions/EntityNotFoundException.cs
using System;

namespace HuHao.BlogService.Domain.Core.Exceptions
{
    public class EntityNotFoundException : DomainException
    {
        public EntityNotFoundException(string message) : base(message)
        {
        }

        public EntityNotFoundException(Type type, IEntityId id) :
            this($"cannot find the {type.Name.ToLower()} with id {id}")
        {
        }
    }
}

File: spring-with-mybatis/src/main/java/study/huhao/demo/domain/core/common/excpetions/EntityNotFoundException.java
package study.huhao.demo.domain.core.common.excpetions;

import study.huhao.demo.domain.core.concepts.Entity;

import java.util.UUID;

public class EntityNotFoundException extends DomainException {
    public EntityNotFoundException(String message) {
        super(message);
    }

    public <T extends Entity> EntityNotFoundException(Class<T> entityClass, UUID id) {
        super("cannot find the " + entityClass.getSimpleName().toLowerCase() + " with id " + id);
    }
}
